lib/mayaUsd/ufe/UsdTransform3dMayaXformStack.cpp

```diff
@@ -0,0 +1,708 @@
+        {TfToken("xformOp:rotateYZX"), toYZX},
+        {TfToken("xformOp:rotateZXY"), toZXY},
+        {TfToken("xformOp:rotateZYX"), toZYX},
+        {TfToken("xformOp:orient"),    nullptr}}; // FIXME, unsupported.
+
+    return cvt.at(opName);
+}
+
+//----------------------------------------------------------------------
+// Conversion functions from all supported rotation attributes to RotXYZ.
+//----------------------------------------------------------------------
+
+typedef Ufe::Vector3d (*CvtRotXYZFromAttrFn)(const VtValue& value);
+
+Ufe::Vector3d fromXYZ(const VtValue& value) {
+    // No rotation order conversion
+    auto v = value.Get<GfVec3f>();
+    return Ufe::Vector3d(v[0], v[1], v[2]);
+}
+
+template<MEulerRotation::RotationOrder SRC_ROT_ORDER>
+Ufe::Vector3d from(const VtValue& value) {
+    auto v = value.Get<GfVec3f>();
+
+    MEulerRotation eulerRot(TO_RAD(v[0]), TO_RAD(v[1]), TO_RAD(v[2]), SRC_ROT_ORDER);
+    eulerRot.reorderIt(MEulerRotation::kXYZ);
+    return Ufe::Vector3d(TO_DEG(eulerRot.x), TO_DEG(eulerRot.y), TO_DEG(eulerRot.z));
+}
+
+auto fromXZY = from<MEulerRotation::kXZY>;
+auto fromYXZ = from<MEulerRotation::kYXZ>;
+auto fromYZX = from<MEulerRotation::kYZX>;
+auto fromZXY = from<MEulerRotation::kZXY>;
+auto fromZYX = from<MEulerRotation::kZYX>;
+
+Ufe::Vector3d fromX(const VtValue& value) {
+    return Ufe::Vector3d(value.Get<float>(), 0, 0);
+}
+
+Ufe::Vector3d fromY(const VtValue& value) {
+    return Ufe::Vector3d(0, value.Get<float>(), 0);
+}
+
+Ufe::Vector3d fromZ(const VtValue& value) {
+    return Ufe::Vector3d(0, 0, value.Get<float>());
+}
+
+CvtRotXYZFromAttrFn getCvtRotXYZFromAttrFn(const TfToken& opName)
+{
+    static std::map<TfToken, CvtRotXYZFromAttrFn> cvt = {
+        {TfToken("xformOp:rotateX"),   fromX},
+        {TfToken("xformOp:rotateY"),   fromY},
+        {TfToken("xformOp:rotateZ"),   fromZ},
+        {TfToken("xformOp:rotateXYZ"), fromXYZ},
+        {TfToken("xformOp:rotateXZY"), fromXZY},
+        {TfToken("xformOp:rotateYXZ"), fromYXZ},
+        {TfToken("xformOp:rotateYZX"), fromYZX},
+        {TfToken("xformOp:rotateZXY"), fromZXY},
+        {TfToken("xformOp:rotateZYX"), fromZYX},
+        {TfToken("xformOp:orient"),    nullptr}}; // FIXME, unsupported.
+
+    return cvt.at(opName);
+}
+
+}
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+namespace {
+
+inline Ufe::Transform3d::Ptr nextTransform3d(
+    const Ufe::Transform3dHandler::Ptr& nextHandler,
+    const Ufe::SceneItem::Ptr&          item
+) {
+    return nextHandler->transform3d(item);
+}
+
+inline Ufe::Transform3d::Ptr nextEditTransform3d(
+    const Ufe::Transform3dHandler::Ptr& nextHandler,
+    const Ufe::SceneItem::Ptr&          item
+) {
+    return nextHandler->editTransform3d(item);
+}
+
+typedef Ufe::Transform3d::Ptr (*NextTransform3dFn)(
+    const Ufe::Transform3dHandler::Ptr& nextHandler,
+    const Ufe::SceneItem::Ptr&          item
+);
+
+Ufe::Transform3d::Ptr createTransform3d(
```

Comment:
Transform3d interface creator for Maya transform stacks.  If the transform op stack doesn't match a Maya stack, give it to the next Transform3dHandler in the chain (at line 259).